framework/graph/GraphBuilder.kt

```diff
@@ -399,26 +465,70 @@ class GraphBuilder(
             }
 
             private fun loadOwnsEdges() {
```

Comment:
There's a fair amount of duplicated code between these two functions, but my attemps to deduplicate this adds more complexity than it removes in my opinion.